q2_fondue/scraper.py

```diff
@@ -189,23 +340,33 @@ def scrape_collection(
             logger.warning(f'Item {attach_key} doesn\'t contain any '
                            f'full-text content')
 
-        # find run IDs
-        run_ids += _find_accession_ids(str_text, 'run')
-        # find bioproject IDs
-        bioproject_ids += _find_accession_ids(str_text, 'bioproject')
+        # find accession (run and bioproject) IDs
+        run_ids = _find_accession_ids(str_text, 'run')
+        bioproject_ids = _find_accession_ids(str_text, 'bioproject')
 
-    # remove duplicate entries in both lists
-    run_ids = list(set(run_ids))
-    bioproject_ids = list(set(bioproject_ids))
+        # match found accession IDs with DOI
+        run_doi = _expand_dict(run_doi, run_ids, doi)
+        bioproject_doi = _expand_dict(
+            bioproject_doi, bioproject_ids, doi)
 
-    if (len(run_ids) == 0) & (len(bioproject_ids) == 0):
+    if (len(run_doi) == 0) & (len(bioproject_doi) == 0):
         raise NoAccessionIDs(f'The provided collection {collection_name} '
                              f'does not contain any run or BioProject IDs')
-    elif len(run_ids) == 0:
+    elif len(run_doi) == 0:
         logger.warning(f'The provided collection {collection_name} '
                        f'does not contain any run IDs')
-    elif len(bioproject_ids) == 0:
+    elif len(bioproject_doi) == 0:
         logger.warning(f'The provided collection {collection_name} '
                        f'does not contain any BioProject IDs')
-    return (pd.Series(run_ids, name='ID'),
-            pd.Series(bioproject_ids, name='ID'))
+
+    # transform dict to dataframe & return
+    run_doi_df = pd.DataFrame.from_dict(
+        run_doi, orient='index', columns=['DOI'])
+    run_doi_df.index.name = 'ID'
+
+    bioproject_doi_df = pd.DataFrame.from_dict(
+        bioproject_doi, orient='index', columns=['DOI'])
+    bioproject_doi_df.index.name = 'ID'
+
+    return (run_doi_df,
+            bioproject_doi_df)
```

Comment:
```suggestion
    dfs = []
    for doi_dict in (run_doi, bioproject_doi):
        df = pd.DataFrame.from_dict(doi_dict, orient='index', columns=['DOI'])
        df.index.name = 'ID'
        dfs.append(df)

    return tuple(dfs)
```

Reply:
this is exactly what I had wanted to do before seeing your comment 😄